lib/spack/spack/operating_systems/windows_os.py

```diff
@@ -31,7 +34,8 @@ class WindowsOs(OperatingSystem):
     10.
     """
 
-    # Find MSVC directories using vswhere
+    # First Strategy: Find MSVC directories using vswhere
+    _compiler_search_paths = []
```

Comment:
I assumes this runs on all operating systems (as part of defining the class) - this didn't specifically happen in this PR, but could you make sure this logic only runs when the class is instantiated (or simpler but slightly hackier: inside of a check for `sys.platform == "win32"`)

Reply:
Made it a property similar to the only other place that demonstrates a use of this attribute, the `CrayFrontend` OS class.